AK/IDAllocator.h

```diff
@@ -26,25 +26,24 @@
 
 #pragma once
 
-#include <stdlib.h>
 #include <AK/HashTable.h>
+#include <stdlib.h>
+#include <time.h>
 
 namespace AK {
 
 class IDAllocator {
 
 public:
-    IDAllocator() {}
-    ~IDAllocator() {}
+    IDAllocator() { srand(time(0)); }
+    ~IDAllocator() { }
 
     int allocate()
     {
         int r = rand();
-        for (int i = 0; i < 100000; ++i) {
+        // Make sure we never vend ID 0, as some code may interpret that as "no ID"
+        for (int i = 1; i < 100000; ++i) {
             int allocated_id = r + i;
-            // Make sure we never vend ID 0, as some code may interpret that as "no ID"
-            if (allocated_id == 0)
-                ++allocated_id;
```

Comment:
This part doesn't make sense to me. What if `r` happens to be `4294967296-i` or slightly lower? Then `r+i` will be `0`. In fact, `i=0` is not a special case at all.

It should take at least 65536 allocated IDs until this loop actually running more than once or twice (see: birthday paradox, and `int` is 32 bit`). Let's not over-optimize this too early. Or do you profile something and IDAllocator was the culprit?